### PR TITLE
[prototype] Restore BC on perspective

### PR DESCRIPTION
**Pull request by datumbox**
This PR is a potential solution for restoring BC on the perspective kernel. 

@vfdev-5 noted that on the new kernels we use the   `perspective_coeffs` parameter which was favoured on the `_FT` and `_FP` stable kernels instead of the `startpoints` and `endpoints` parameters exposed by stable `F`. This would be BC breaking and thus we have to restore the original parameters.

Unfortunately the estimation of `perspective_coeffs` is quite [computationally expensive](https://github.com/pytorch/vision/blob/bdc55567d0e0f639b1c7b1dc4374819bd1b9693f/torchvision/transforms/functional.py#L662-L687) and thus it would be great if we can estimate them [once](https://github.com/pytorch/vision/blob/70faba91d1c79f689ac6ebd30ad0c4be62690196/torchvision/prototype/transforms/_geometry.py#L526) per record on the `RandomPerspective` transform and share it with all the inputs. A middle ground is to introduce a `coefficients` parameters on all the `perspective*` kernels and dispatchers which will allow users to pass directly the coefficients if they have them.

This PR:
- Restores the `startpoints` and `endpoints` parameters on the original positional locations of `perspective*` methods.
- Introduces an extra keyword argument called `coefficients` in all the `perspective*` methods. 
- Changes the types of all above parameters to make it possible to accept `None` values. Either `startpoints/endpoints` or `coeffiecients` must be non-`None`.
- Updates the tests to pass `startpoints=None, endpoints=None` and `coeffiecients=perspective_coeffs`. 

Might be worth on a follow up to add tests for the case where `startpoints/endpoints` are non-None.

cc @vfdev-5 @bjuncek @pmeier

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks for quick fix, @datumbox !

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 1]
```diff
@@ -906,12 +907,30 @@ def crop(inpt: features.InputTypeJIT, top: int, left: int, height: int, width: i
...
+    coefficients: Optional[List[float]],
+) -> List[float]:
+    if coefficients is not None:
+        if len(coefficients) != 8:
+            raise ValueError("Argument coefficients should have 8 float values")
+        return coefficients
+    elif startpoints is not None and endpoints is not None:
+        return _get_perspective_coeffs(startpoints, endpoints)
+    else:
+        raise ValueError("Either the startpoints/endpoints or the coefficients must have non `None` values.")
```
Comment (pmeier) [inline torchvision/prototype/transforms/functional/_geometry.py:922]: Shouldn't we also check for both variants passed? I think passing passing all three parameters and ignoring  `startpoints` / `endpoints` is not good UX.

File: test/prototype_transforms_kernel_infos.py [hunk 2]
```diff
@@ -1184,38 +1184,38 @@ def reference_inputs_pad_bounding_box():
 def sample_inputs_perspective_image_tensor():
     for image_loader in make_image_loaders(sizes=["random"]):
         for fill in [None, 128.0, 128, [12.0], [12.0 + c for c in range(image_loader.num_channels)]]:
-            yield ArgsKwargs(image_loader, fill=fill, perspective_coeffs=_PERSPECTIVE_COEFFS[0])
+            yield ArgsKwargs(image_loader, None, None, fill=fill, coefficients=_PERSPECTIVE_COEFFS[0])
```
Comment (pmeier) [inline test/prototype_transforms_kernel_infos.py:1187]: Nit: While correct, can we pass keywords here to make this call more readable?

```suggestion
            yield ArgsKwargs(image_loader, startpoints=None, endpoints=None, fill=fill, coefficients=_PERSPECTIVE_COEFFS[0])
```

File: torchvision/prototype/transforms/functional/_geometry.py (hunk 1 above)
Comment (datumbox) [reply torchvision/prototype/transforms/functional/_geometry.py:922]: Sounds good, I'll make the change.

File: test/prototype_transforms_kernel_infos.py (hunk 2 above)
Comment (datumbox) [reply test/prototype_transforms_kernel_infos.py:1187]: Happy to make this change on the tests. I wouldn't want to make it on the main code because the arguments are positional not keyword args. Any concerns on that?

Comment (pmeier) [reply test/prototype_transforms_kernel_infos.py]: Nono, fine with that. The test definition is just hard to read without knowledge of the exact signature of the kernel.

Comment (datumbox) [reply test/prototype_transforms_kernel_infos.py:1187]: Something, somewhere in the testing framework seems to send [multiple parameters](https://github.com/pytorch/vision/actions/runs/3392776645/jobs/5639353266) to the kernel. I don't want to waste too much time on this, I've reverted the named params change and we can do the optimization on a follow up as it doesn't affect the main implementation or the quality of the tests.
